engine/resource/src/async/load_queue_threaded.cpp

```diff
@@ -213,11 +207,18 @@ namespace dmLoadQueue
 
         uint32_t old_bytes_waiting = queue->m_BytesWaiting;
 
+        uint32_t buffer_capacity = request->m_Buffer.Capacity();
+        queue->m_BytesWaiting -= buffer_capacity;
```

Comment:
The preloader system is quite intricate, and I had missed that the buffer size was also used to "unlock" the next item in the thread queue.